lms/djangoapps/verify_student/views.py

```diff
@@ -1047,3 +1049,82 @@ def reverification_window_expired(_request):
     """
     # TODO need someone to review the copy for this template
     return render_to_response("verify_student/reverification_window_expired.html")
+
+
+class ICRVReverifyView(View):
+    """
+    The in-course reverification view.
+    Needs to perform these functions:
+        - take new face photo
+        - retrieve the old id photo
+        - submit these photos to photo verification service
+
+    Does not need to worry about pricing
+    """
+    @method_decorator(login_required)
+    def get(self, request, course_id, checkpoint_name):
+        """
+        display this view
+        """
+        user = request.user
+        course_id = CourseKey.from_string(course_id)
+        course = modulestore().get_course(course_id)
+        if course is None:
+            raise Http404
+        checkpoint = VerificationCheckpoint.get_verification_checkpoint(course_id, checkpoint_name)
+        if checkpoint is None:
+            raise Http404
+        try:
+            SoftwareSecurePhotoVerification.original_verification(user)
+        except IndexError:
+            path = reverse('verify_student_verify_later', kwargs={'course_id': unicode(course_id)})
+            return redirect(path)
+        context = {
+            "user_full_name": request.user.profile.name,
+            "error": False,
+            "course_id": course_id.to_deprecated_string(),
+            "course_name": course.display_name_with_default,
+            "course_org": course.display_org_with_default,
+            "course_num": course.display_number_with_default,
+            "checkpoint_name": checkpoint_name,
+        }
+
+        return render_to_response("verify_student/midcourse_photo_reverification.html", context)
+
+    @method_decorator(login_required)
+    def post(self, request, course_id, checkpoint_name):
+        """
+        submits the reverification to SoftwareSecure
+        """
+        try:
+            user = request.user
+            now = datetime.datetime.now(UTC)
+            course_id = CourseKey.from_string(course_id)
+            checkpoint = VerificationCheckpoint.get_verification_checkpoint(course_id, checkpoint_name)
+            if checkpoint is None:
+                raise VerificationCheckpointException
+            # TODO: How to get this window for new implementation. I think window is in VerificationBlock
+            window = MidcourseReverificationWindow.get_window(course_id, now)
+            attempt = SoftwareSecurePhotoVerification(user=request.user, window=window)
```

Comment:
The query here should retrieve the last submitted or successful attempt.  Something like:
```
SoftwareSecurePhotoVerification.objects.get(user=request.user, status__in=["submitted", "approved").order_by("-created_at")
```
We should do the same check in the GET request as well.  At this point, if the user hasn't submitted a verification, we should redirect them to the initial verification flow.